tsconfig.json

```diff
@@ -1,8 +1,8 @@
 {
   "compilerOptions": {
     "declaration": true,
-    "declarationDir": "./dist",
-    "outDir": "./dist",
+    "declarationDir": "./lib",
```

Comment:
Is it normal that the generated files for the type definition are in the `dist` folder ?
I don't have 'lib'  generated folder.

Reply:
Files are generated in the `lib` directory only when running with the `tsc` command or build triggered by IDE (like IntelliJ).
When running with rollup, the typescript plugin seems generating the definition files in the same folder as the one where the bundle is generated, so the `dist` folder.

Reply:
I have changed the tsconfig.json because I would like we provide an non bundled version of the ESM package. We could use the lib folder to package it in the npm package.

Reply:
For me, it's weird to have the generated files not at the same place if we us `rollup` or `tsc`.

@aibcmars A idea ?

Reply:
at first glance it is strange indeed
I have started to check it up locally - should be able to tell something more soon

Reply:
Just remember that rollup generates bundles (single javascript files with all the lib) and the purpose of this PR is to split the output directories based on usage of the generated resources. Mainly to avoid the current mess we have and which requires cleaning: we have everything in the dist folder, we don't know where the files come from (see commit and PR description).

The dev bundle for the demo/development is now generated in a subfolder of the build and the dev http server uses it.
The bundles for the npm package stay in the dist folder to avoid package breaking change.
We have each javascript generated files from each typescript files in the lib folder. We may in the future use them directly to generate the bundles without having to rebuild them with the typescript plugin. This is what roughjs does for instance: https://github.com/rough-stuff/rough/blob/v3.1.0/package.json#L10

Reply:
split is a great idea 👍 
bundles ok to have it under /dist 👍 

however, I do not really see the interest in having the additional /lib directory

Reply:
I am still wondering?
[in the case for not bundled esm]

Isn't it enough to just have the definitions under /dist

Reply:
We could talk about this directly on next week.

As having a lib folder may only be useful in case of a non bundled ESM module (not done for now), we could generate the single javascript files in `build/lib` or `build/bin` to hide them.
For the declaration files, we can store them wherever we want as far as we reference the right path in the package.json `types` field.

Reply:
We (@csouchet, @aibcmars and @tbouffard) have decided to
- store the tsc outputs to `build/lib`
- don't generate declarations when running tsc as they are never used